Viktor's readme.tex

```diff
@@ -0,0 +1,74 @@
+
+An initial implementation will provide each cell with the longevity of a collection of neighbouring cells.
+%
+The cell may also inspect the behaviour of the neighbouring cells and at each time-step the cell randomly samples a neighbour weighted by longevity, whose behaviour it will train itself to mimic.
+%
+If the sampled neighbour died during the time-step the cell instead trains itself to avoid the behaviour exhibited by the neighbour.
+
+The system can be further generalised where the cell isn't provided the longevity of the neighbouring cells by the system but where each cell instead learns to estimate the fitness of others.
+%
+This fitness is then used in place of the system-provided longevity score when selecting whose behaviour to mimic.
+
+\section{Competitive Environments and Group Dynamics}
+
+When placed in a competitive environment with other cells where the cells compete for the acquirement of resources or where interference competition can utilised to ensure longevity of its own lineage at the expense of other cells, the cell may also use ability to inspect the behaviour of its neighbours to better determine its own behaviour.
+
+Since each neighbour also has the ability to inspect the behaviour of the cell it is not always optimal to exhibit optimal behaviour at each time-step.
+%
+It is of interest to study whether the behaviour of the cells will converge to strategies that not only greedily solve the task but to conservative strategies where the cell is less susceptible to competition from its neighbours.
+
+A cell may also be constructed from a collection of sub-cells that collectively determine the behaviour of the cell.
+
+\section{Shared Genetics and General-Purpose Modules}
+
+The genetic make-up of the cell, the parameters that determine its behaviour, can be shared between the cells in such a way that training the parameters to optimise for one cell also implicitly updates the parameters for all other cells the parameters are shared with.
+%
+Another form of competition may arise from this sharing of parameters.
+%
+By making this module of shared parameters better the cell also increases the fitness of its competition resulting in a kind of self-competition where it's not always beneficial to maximise its own ability.
+
+Moreover, if the shared module is just a subset of the make-up of the cells then this module has to adapt to solving the multiple goals of the differing cells.
+%
+Whether or not this leads to more general-purpose modules or whether the module becomes useless due to excessive competition of the optimisation of the parameters could be interesting to look into.
+
+Conversely, if the shared module constitutes the entirety of the parameters of the cell, the cells that share this module may be forced to cooperate if the downsides of competing outweigh the benefits of following a strategy that leads to cooperation with the cells with shared parameters.
+
+\section*{outline?}
```

Comment:
This was not intended to be included in the document. It was a comment I left in to remind myself of what to write. I think we should just drop this list from the document.